grails-app/domain/org/pih/warehouse/invoice/InvoiceItem.groovy

```diff
@@ -58,7 +59,7 @@ class InvoiceItem implements Serializable {
     User createdBy
     User updatedBy
 
-    static belongsTo = [invoice: Invoice]
+    static belongsTo = [Invoice, ShipmentItem, OrderAdjustment, OrderItem]
```

Comment:
This worries me a tiny bit. Do we know what impact this is going to have?

Reply:
@jmiranda could you elaborate here on why this worries you? I had to put it here in order to be able to add it in hasMany relation on OrderItem, ShipmentItem, and OrderAdjustment (which is crucial for this performance and code overall improvements)

Reply:
It could generate more SQL than we're expecting and could consequently have unintended consequences on performance. 

I think you should be able to define invoice item with invoice as a parent and hasMany on other domains without the need to define cascade behavior (which is what belongsTo is doing). 

https://stackoverflow.com/questions/23905638/grails-domain-class-hasone-hasmany-without-belongsto

I assume you're encountering a Grails error without this. So post that here.

Reply:
To resolve this just let me know 

1. if it's possible to include back references on each of these. And if not, what is the error you're getting from Grails. 
1. Also does this add any unnecessary SQL queries when retrieving invoice items. Does it go through each of those associations (in either case, with or without back reference).